test/unit/validate_test.lua

```diff
@@ -432,21 +431,85 @@ local error_cases = {
         },
         err = "http endpoint metrics 'enabled' must be a boolean, got string",
     },
+    ["integration_with_httpd_role_server_is_not_string"] = {
+        cfg = { http = { { server = 1 } } },
+        err = "server configuration sould be a string, got number",
+    },
+    ["integration_with_httpd_role_server_and_listen_simultaneously"] = {
+        cfg = {
+            http = {
+                {
+                    server = "additional",
+                    listen = 8001,
+                }
+            }
+        },
+        err = "it is not possible to provide 'server' and 'listen' blocks simultaneously",
+    },
+    ["integration_with_httpd_role_cfg_missing"] = {
+        cfg = { http = { { server = "additional" } } },
+        mocks = {
+            {
+                module = "config",
+                method = "get",
+                implementation = function ()
+                    return {}
+                end,
+            },
+        },
+        err = "there is no configuration for httpd role",
+    },
+    ["integration_with_httpd_role_server_not_found"] = {
+        cfg = { http = { { server = "not_existing_server" } } },
+        mocks = {
+            {
+                module = "config",
+                method = "get",
+                implementation = function (_, param)
+                    if param == "roles_cfg" then
+                        return {
+                            ['roles.httpd'] = httpd_config,
+                        }
+                    end
+                    return {}
+                end,
+            },
+        },
+        err = "server with name not_existing_server not found in httpd role config",
+    },
 }
 
 for name, case in pairs(error_cases) do
     g["test_validate_error_" .. name] = function(gc)
+        if case.mocks ~= nil then
+            mocks.init(case.mocks)
+            mocks.apply()
+        end
+
         t.assert_error_msg_contains(case.err, function()
             gc.role.validate(case.cfg)
         end)
+
+        if case.mocks ~= nil then
+            mocks.delete()
```

Comment:
The name is a bit confusing.
```suggestion
            mocks.clear()
```

Reply:
Edited.